riklet/src/runtime/function_runtime.rs

```diff
@@ -64,17 +55,43 @@ impl Runtime for FunctionRuntime {
             network_interfaces: vec![NetworkInterface {
                 iface_id: "eth0".to_string(),
                 guest_mac: Some(generate_mac_addr().to_string()),
-                host_dev_name: self.network.tap_name().map_err(RuntimeError::NetworkError)?,
+                host_dev_name: self
+                    .network
+                    .tap_name()
+                    .map_err(RuntimeError::NetworkError)?,
             }],
         });
 
-        event!(Level::DEBUG, "Starting the MicroVM");
+        Ok(config)
+    }
+}
+
+#[async_trait]
+impl Runtime for FunctionRuntime {
+    #[tracing::instrument(skip(self), fields(host_iface = %self.network.tap_name().unwrap()))]
+    async fn run(&mut self) -> Result<()> {
+        self.network
+            .init()
+            .await
+            .map_err(RuntimeError::NetworkError)?;
+
+        event!(Level::DEBUG, "Define network");
+
+        let firecracker = Firecracker::new(firepilot::FirecrackerOptions {
+            command: Some(self.function_config.firecracker_location.clone()),
+            ..Default::default()
+        })
+        .map_err(RuntimeError::FirecrackerError)?;
+
+        event!(Level::DEBUG, "Generate configuration for MicroVM");
+        let vm = self.generate_microvm_config()?;
```

Comment:
One part of this pr is to improve readability so here maybe use `vm_config` instead of juste `vm`